nautobot/extras/api/serializers.py

```diff
@@ -569,6 +570,21 @@ class Meta:
         model = JobQueue
         fields = "__all__"
 
+    def validate(self, data):
+        if self.instance:
+            job_queue_type = data.get("queue_type", self.instance.queue_type)
+            external_integration = data.get("external_integration", self.instance.external_integration)
+
+            if external_integration and job_queue_type != JobQueueTypeChoices.TYPE_KUBERNETES:
+                error_message = f"Only job queues with type {JobQueueTypeChoices.TYPE_KUBERNETES} can be linked to external integration objects."
+                errors = {}
+                if "external_integration" in data:
+                    errors["external_integration"] = [error_message]
```

Comment:
I wonder if this error shouldn't be assigned to the `job_queue_type` as well?

Reply:
Good call I had the error assigned to `job_queue_type` as well before but I deleted that part because I thought it was redundant. I can add it back